src/preset.ts

```diff
@@ -1180,7 +1055,64 @@ async function tryApplyVsDevEnv(preset: ConfigurePreset, workspaceFolder: string
     preset.__parentEnvironment = EnvironmentUtils.mergePreserveNull([process.env, developerEnvironment]);
 }
 
-async function expandConfigurePresetHelper(folder: string, preset: ConfigurePreset, allowUserPreset: boolean = false, errorHandler?: ExpansionErrorHandler) {
+export async function expandConfigurePreset(folder: string, name: string, workspaceFolder: string, sourceDir: string, allowUserPreset: boolean = false, enableTryApplyDevEnv: boolean = true, errorHandler?: ExpansionErrorHandler): Promise<ConfigurePreset | null> {
+
+    // TODO: We likely need to refactor to include these refs, for configure, build, test, etc Presets.
+    const refs = referencedConfigurePresets.get(folder);
+    if (!refs) {
+        referencedConfigurePresets.set(folder, new Set());
+    } else {
+        refs.clear();
+    }
+
+    // if enableTryApplyDevEnv is true, we have to check if the preset is valid in expandedPresetsFiles
+    if (enableTryApplyDevEnv) {
+        const preset = getPresetByName(configurePresets(folder), name);
+        if (!preset) {
+            return null;
+        }
+    }
+
+    const preset = await expandConfigurePresetImpl(folder, name, allowUserPreset, enableTryApplyDevEnv, errorHandler);
+
+    if (!preset) {
+        return null;
+    }
+
+    if (enableTryApplyDevEnv) {
+        // modify the preset parent environment, in certain cases, to apply the Vs Dev Env on top of process.env.
+        await tryApplyVsDevEnv(preset, workspaceFolder, sourceDir);
+    }
+
+    const expandedPreset = await expandConfigurePresetVariables(preset, folder, name, workspaceFolder, sourceDir, allowUserPreset, enableTryApplyDevEnv, errorHandler);
+
+    errorHandlerHelper(preset.name, errorHandler);
+
+    // Other fields can be copied by reference for simplicity
+    merge(expandedPreset, preset);
+
+    return expandedPreset;
+}
+
+async function expandConfigurePresetImpl(folder: string, name: string, allowUserPreset: boolean = false, useOriginalPresets: boolean = false, errorHandler?: ExpansionErrorHandler): Promise<ConfigurePreset | null> {
```

Comment:
We should add a comment explaining the `useOriginalPresets` variable here so that people understand the use case and why we have it. It's nice to have this functionality overall, but right now it's purely for an edge case, so it might even be worthwhile to have the edge case described in short in the comment for this function.